go/filemonitor/filelistener.go

```diff
@@ -0,0 +1,160 @@
+package filemonitor
+
+import (
+	"bufio"
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	slog "github.com/burke/zeus/go/shinylog"
+)
+
+type fileListener struct {
+	gatheringMonitor
+	netListener net.Listener
+	connections map[net.Conn]chan string
+	stop        chan struct{}
+	sync.Mutex
+	wg sync.WaitGroup
+}
+
+func NewFileListener(ln net.Listener) FileMonitor {
+	fl := fileListener{
+		netListener: ln,
+		connections: make(map[net.Conn]chan string),
+		stop:        make(chan struct{}),
+	}
+	fl.changes = make(chan string)
+
+	go fl.serveListeners()
+	go fl.serve()
+
+	return &fl
+}
+
+func (f *fileListener) Add(file string) error {
+	f.Lock()
+	defer f.Unlock()
+
+	for _, ch := range f.connections {
+		ch <- file
+	}
+
+	return nil
+}
+
+func (f *fileListener) Close() error {
+	f.Lock()
+
+	select {
+	case <-f.stop:
+		f.Unlock()
+		return nil // Already stopped
+	default:
+		close(f.stop)
+	}
+
+	var firstErr error
+	if firstErr = f.netListener.Close(); firstErr != nil {
+		slog.Trace("Error closing file listener: %v", firstErr)
```

Comment:
If you pass `--log` to Zeus, it logs them to whatever file you provide. In our setup, you can read these trace logs in `dev/zeus.log`... They're pretty helpful for debugging now that I found them!